src/subcommand.rs

```diff
@@ -107,22 +52,32 @@ impl Subcommand {
         let config = LocalizedConfig::find_cargo_config_for_workspace(&root_dir)?;
 
         let target_dir = target_dir.unwrap_or_else(|| {
-            utils::find_workspace(&manifest, &package)
+            utils::find_workspace(&manifest_path, &package)
                 .unwrap()
-                .unwrap_or_else(|| manifest.clone())
+                .unwrap_or_else(|| manifest_path.clone())
                 .parent()
                 .unwrap()
                 .join(utils::get_target_dir_name(config.as_deref()).unwrap())
         });
-        if examples {
+
+        let mut artifacts = vec![];
+        if args.examples {
             for file in utils::list_rust_files(&root_dir.join("examples"))? {
                 artifacts.push(Artifact::Example(file));
             }
+        } else {
+            for example in &args.example {
+                artifacts.push(Artifact::Example(example.into()));
+            }
         }
-        if bins {
+        if args.bins {
             for file in utils::list_rust_files(&root_dir.join("src").join("bin"))? {
                 artifacts.push(Artifact::Root(file));
             }
+        } else {
+            for bin in &args.bin {
+                artifacts.push(Artifact::Root(bin.into()));
+            }
         }
         if artifacts.is_empty() {
             artifacts.push(Artifact::Root(package.clone()));
```

Comment:
Since we have the manifest available in scope here, I feel like we should process `[lib] name = ""` renames (similar to the `[[bin]]` comment above) and `autolib = <bool>` here?